### PR TITLE
[search] Call Emiter::Finish for bookmarks search from bookmarks processor, not geocoder.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
JTALL

File: search/bookmarks/processor.cpp [hunk 1]
```diff
@@ -271,6 +271,8 @@ void Processor::Search(Params const & params) const
   }
 }
 
+void Processor::Finish(bool canceled) { m_emitter.Finish(canceled); }
```
Comment (mpimenov) [inline search/bookmarks/processor.cpp]: s/canceled/cancelled/

File: search/processor.cpp [hunk 2]
```diff
@@ -531,12 +534,22 @@ void Processor::SearchPlusCode()
...
   InitParams(params);
   params.m_groupId = groupId;
-  m_bookmarksProcessor.Search(params);
+  try
+  {
+    m_bookmarksProcessor.Search(params);
+  }
+  catch (CancelException const &)
+  {
+    LOG(LDEBUG, ("Search has been cancelled."));
```
Comment (mpimenov) [inline search/processor.cpp]: s/Search/Bookmarks search/ ?

File: search/bookmarks/processor.cpp (hunk 1 above)
Comment (tatiana-yan) [reply search/bookmarks/processor.cpp]: done

File: search/processor.cpp (hunk 2 above)
Comment (tatiana-yan) [reply search/processor.cpp]: done

File: search/processor.cpp [hunk 3]
```diff
@@ -531,12 +534,22 @@ void Processor::SearchPlusCode()
...
 }
 
-void Processor::SearchBookmarks(bookmarks::GroupId const & groupId) const
+void Processor::SearchBookmarks(bookmarks::GroupId const & groupId)
 {
   bookmarks::Processor::Params params;
   InitParams(params);
   params.m_groupId = groupId;
-  m_bookmarksProcessor.Search(params);
+  try
```
Comment (mpimenov) [inline search/processor.cpp]: empty line before ```try```, please

Comment (tatiana-yan) [reply search/processor.cpp]: done

**Pull request merged**